### PR TITLE
Fix/voro beta

### LINKED ISSUE #506: Voronoi beta
1. Wrapping position for plotting: positions of Voronoi vertices are not matched with the whole box.
position_wrapping

2. Wrong ordering of vertices gives a 2D plot of patches with inner lines.
current_voronoi_plot

3. The doc for Voronoi needs to be updated including adding in overview.

### PR DESCRIPTION
**Pull request by bdice**
## Description
Switching to `voro++` and rewriting the Voronoi class left a few bugs. I fixed those and added some features that improve the plotting code for examples.

## Motivation and Context
Resolves: #506.

## How Has This Been Tested?
Rewrote example notebook to test this. Existing tests pass.

## Screenshots
(To be included in a future version of the examples)
image

## Types of changes

- [ ] Bug fix (non-breaking change which fixes an issue)
- [x] New feature (non-breaking change which adds or improves functionality)
- [x] Breaking change (fix or feature that would cause existing functionality to change)

## Checklist:

- [x] I have read the [**CONTRIBUTING**](https://github.com/glotzerlab/freud/blob/master/CONTRIBUTING.md) document.
- [x] My code follows the code style of this project.
- [x] I have updated the documentation (if relevant).
- [x] I have added tests that cover my changes (if relevant).
- [x] All new and existing tests passed.
- [x] I have updated the [credits](https://github.com/glotzerlab/freud/blob/master/doc/source/credits.rst).
- [x] I have updated the [Changelog](https://github.com/glotzerlab/freud/blob/master/ChangeLog.md).

### CODE REVIEW

File: cpp/locality/Voronoi.cc [hunk 1]
```diff
@@ -97,9 +97,30 @@ void Voronoi::compute(const freud::locality::NeighborQuery* nq)
...
+
+                // Sort relative vertices by their angle in 2D systems
+                if (box.is2D())
+                {
+                    std::sort(relative_vertices.begin(), relative_vertices.end(),
+                            [](const vec3<double> a, const vec3<double> b)
+                            {
+                                const double a_angle = std::atan2(a.y, a.x);
+                                const double b_angle = std::atan2(b.y, b.x);
+                                return a_angle < b_angle;
```
Comment (vyasr) [inline cpp/locality/Voronoi.cc]: I'd probably just write this as a one-liner, though I can't imagine any modern compiler actually cares about the difference.

File: cpp/locality/Voronoi.cc [hunk 2]
```diff
@@ -97,9 +97,30 @@ void Voronoi::compute(const freud::locality::NeighborQuery* nq)
...
+                            });
+                }
+
+                // Save polytope vertices in system coordinates
+                std::vector<vec3<double> > system_vertices;
+                for (auto vertex_iter = relative_vertices.begin(); vertex_iter != relative_vertices.end(); vertex_iter++)
+                {
+                    vec3<double> query_point((*nq)[query_point_id]);
+                    system_vertices.push_back((*vertex_iter) + query_point);
```
Comment (vyasr) [inline cpp/locality/Voronoi.cc:118]: Can't you just use `ri` directly?

A potential simplification/microoptimization would be to get rid of the calls to `voronoi_loop.[xyz]` and just set `const vec3<double> &ri = nq[query_point_id]`.

Comment (vyasr) [reply cpp/locality/Voronoi.cc:118]: If you did that, I'd probably just rename the variable to `query_point`.

File: doc/source/locality.rst [hunk 3]
```diff
@@ -7,11 +7,16 @@ Locality Module
...
+    freud.locality.LinkCell
     freud.locality.NeighborList
     freud.locality.NeighborQuery
     freud.locality.NeighborQueryResult
-    freud.locality.AABBQuery
-    freud.locality.LinkCell
+    freud.locality.PairCompute
+    freud.locality.RawPoints
+    freud.locality.SpatialHistogram
+    freud.locality.SpatialHistogram1D
```
Comment (vyasr) [inline doc/source/locality.rst:18]: We probably should hide about half of these, but we can do that in a separate PR that properly renames them everywhere.

**Review by vyasr — APPROVED** [review]
I have a few change suggestions, but they're minor so feel free to act on them or not as you see fit. The changes look good to me.

File: doc/source/locality.rst (hunk 3 above)
Comment (bdice) [reply doc/source/locality.rst:18]: I created #512.

**Pull request merged**